src/bin/init.rs

```diff
@@ -45,16 +47,28 @@ pub fn execute(options: Options, config: &Config) -> CliResult<Option<()>> {
                           options.flag_frozen,
                           options.flag_locked));
 
-    let Options { flag_bin, arg_path, flag_name, flag_vcs, .. } = options;
+    let Options { flag_bin, flag_lib, arg_path, flag_name, flag_vcs, .. } = options;
+
+    let is_lib = if !flag_bin {
+        true
+    }
+    else {
+        flag_lib
+    };
 
     let opts = ops::NewOptions {
         version_control: flag_vcs,
         bin: flag_bin,
+        lib: is_lib,
         path: &arg_path.unwrap_or(format!(".")),
         name: flag_name.as_ref().map(|s| s.as_ref()),
     };
 
     try!(ops::init(opts, config));
+
+    try!(config.shell().status("Created", format!("{} project",
+                                                   if is_lib { "library" }
+                                                   else {"binary (application)"})));
```

Comment:
Oh ok, in that case maybe it could be pushed down into the `init` and `new` functions? (would be good to deduplicate the `if` above in any case)